android/expoview/src/main/java/host/exp/exponent/notifications/ScopedNotificationsIdUtils.java

```diff
@@ -7,34 +7,41 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.annotation.RequiresApi;
+import expo.modules.notifications.notifications.model.NotificationCategory;
 import host.exp.exponent.kernel.ExperienceId;
 
-public class ScopedNotificationsChannelUtils {
+public class ScopedNotificationsIdUtils {
   static final String SCOPED_GROUP_TAG = "EXPO_GROUP";
   static final String SCOPED_CHANNEL_TAG = "EXPO_CHANNEL";
+  static final String SCOPED_CATEGORY_TAG = "EXPO_CATEGORY";
   static final String SEPARATOR = "/";
 
   @NonNull
-  static String getScopedChannelId(@NonNull ExperienceId experienceId, @NonNull String channelId) {
+  public static String getScopedChannelId(@NonNull ExperienceId experienceId, @NonNull String channelId) {
     return SCOPED_CHANNEL_TAG + SEPARATOR + experienceId.get() + SEPARATOR + channelId;
   }
 
   @NonNull
-  static String getScopedGroupId(@NonNull ExperienceId experienceId, @NonNull String channelId) {
+  public static String getScopedGroupId(@NonNull ExperienceId experienceId, @NonNull String channelId) {
     return SCOPED_GROUP_TAG + SEPARATOR + experienceId.get() + SEPARATOR + channelId;
   }
 
+  @NonNull
+  public static String getScopedCategoryId(@NonNull ExperienceId experienceId, @NonNull String categoryId) {
+    return SCOPED_CATEGORY_TAG + SEPARATOR + experienceId.get() + SEPARATOR + categoryId;
+  }
+
   @Nullable
-  static String getUnscopedId(@Nullable String scopedId) {
-    if (scopedId == null || !scopedId.startsWith(SCOPED_CHANNEL_TAG) && !scopedId.startsWith(SCOPED_GROUP_TAG)) {
+  public static String getUnscopedId(@Nullable String scopedId) {
+    if (scopedId == null || !scopedId.startsWith(SCOPED_CHANNEL_TAG) && !scopedId.startsWith(SCOPED_GROUP_TAG) && !scopedId.startsWith(SCOPED_CATEGORY_TAG)) {
```

Comment:
Frankly speaking I'm usually afraid of unparanthersised `… || … && …`, but if it works… 👍